go/kbfs/libkbfs/folder_branch_ops.go

```diff
@@ -901,15 +914,50 @@ func (fbo *folderBranchOps) syncOneNode(
 	}
 }
 
+func (fbo *folderBranchOps) startOp(
+	ctx context.Context, fs string, args ...interface{}) (
+	time.Time, *time.Timer) {
+	now := fbo.config.Clock().Now()
+	// Immediately log with vlog, and if the operation takes more than
+	// one second, log via the normal logger too.
+	fbo.deferVlog.CLogf(ctx, libkb.VLog1, fs, args...)
+	timer := time.AfterFunc(1*time.Second, func() {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			newArgs := make([]interface{}, len(args)+1)
+			newArgs[0] = now
+			copy(newArgs[1:], args)
+			fbo.deferLog.CDebugf(
+				ctx, "(Long operation, started=%s) "+fs, newArgs...)
+		}
+	})
```

Comment:
Do you have any optimizations in mind?

I was fine with this closure since it's only invoked in the rare case that an operation takes more than a second (at which point perf doesn't matter anymore).  I don't know if _constructing_ the closure is itself expensive though.

Which extra goroutine are you talking about?  I'm not positive but I don't think `time.AfterFunc` makes one until the timer expires.  For example: https://play.golang.org/p/rgJvhGMuWr3

Reply:
Hmm how does it implement a timer without a goroutine? Interesting...

Reply:
I guess it's some fancy assembly-level queue that registers for the shortest timer left, and calls the timer function when that one fires.  I looked through the Go source and hit an asm wall at some point.

Reply:
Neat! Ok leave it then.